src/Libraries/Microsoft.Extensions.AI.Abstractions/Functions/AIFunctionMetadata.cs

```diff
@@ -76,33 +65,15 @@ public string Description
         init => _description = value ?? string.Empty;
     }
 
-    /// <summary>Gets the metadata for the parameters to the function.</summary>
-    /// <remarks>If the function has no parameters, the returned list is empty.</remarks>
-    public IReadOnlyList<AIFunctionParameterMetadata> Parameters
-    {
-        get => _parameters;
-        init => _parameters = Throw.IfNull(value);
-    }
-
-    /// <summary>Gets the <see cref="AIFunctionParameterMetadata"/> for a parameter by its name.</summary>
-    /// <param name="name">The name of the parameter.</param>
-    /// <returns>The corresponding <see cref="AIFunctionParameterMetadata"/>, if found; otherwise, null.</returns>
-    public AIFunctionParameterMetadata? GetParameter(string name)
-    {
-        Dictionary<string, AIFunctionParameterMetadata>? parametersByName = _parametersByName ??= _parameters.ToDictionary(p => p.Name);
-
-        return parametersByName.TryGetValue(name, out AIFunctionParameterMetadata? parameter) ?
-            parameter :
-            null;
-    }
-
-    /// <summary>Gets parameter metadata for the return parameter.</summary>
-    /// <remarks>If the function has no return parameter, the value is a default instance of an <see cref="AIFunctionReturnParameterMetadata"/>.</remarks>
-    public AIFunctionReturnParameterMetadata ReturnParameter
-    {
-        get => _returnParameter;
-        init => _returnParameter = Throw.IfNull(value);
-    }
+    /// <summary>
+    /// Gets a <see cref="MethodInfo"/> for the underlying .NET method this <see cref="AIFunction"/> represents.
+    /// </summary>
+    /// <remarks>
+    /// This property provides additional metadata on the function and its signature.
+    /// Setting this property is optional and should have no impact on function invocation or its JSON schema,
+    /// which is how <see cref="IChatClient"/> implementations interface with AI functions primarily.
+    /// </remarks>
+    public MethodInfo? UnderlyingMethod { get; init; }
```

Comment:
Without yet fully understanding the design intent, I'm not clear on this from the XML docs alone. Why would someone set an `UnderlyingMethod` if it "has no impact on function invocation or its JSON schema"? Or as a consumer, when should I expect this to be available or not available?

Reply:
Does this look better?
```
    /// <summary>
    /// Gets the underlying <see cref="MethodInfo"/> that this <see cref="AIFunction"/> might be wrapping.
    /// </summary>
    /// <remarks>
    /// Provides additional metadata on the function and its signature. Implementations not wrapping .NET methods may return <see langword="null"/>.
    /// </remarks>
    public virtual MethodInfo? UnderlyingMethod { get; init; }
```